crates/turborepo-lib/src/run/scope/change_detector.rs

```diff
@@ -99,15 +100,24 @@ impl<'a> GitChangeDetector for ScopeChangeDetector<'a> {
             .change_mapper
             .changed_packages(changed_files, lockfile_contents)?
         {
-            PackageChanges::All => Ok(self
-                .pkg_graph
-                .packages()
-                .map(|(name, _)| name.to_owned())
-                .collect()),
-            PackageChanges::Some(packages) => Ok(packages
-                .iter()
-                .map(|package| package.name.to_owned())
-                .collect()),
+            PackageChanges::All => {
+                debug!("all packages changed");
```

Comment:
related improvement to this in https://github.com/vercel/turbo/pull/8872 (i can rebase my PR on top of this branch!)